src/App.tsx

```diff
@@ -1,25 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [page, setPage] = useState(1);
+
+  const visibleItems = (arrOfItems: string[]) => {
+    const res = [];
+
+    if (Math.floor(42 / page) < perPage) {
+      for (let i = perPage * page - perPage; i < 42; i++) {
+        res.push(arrOfItems[i]);
+      }
+    } else {
+      for (let i = perPage * page - perPage; i < perPage * page; i++) {
+        res.push(arrOfItems[i]);
+      }
+    }
+
+    return res;
+  };
+
+  const shownItems = visibleItems(items);
+
+  const handlePage = (chosenPage: number) => {
+    setPage(chosenPage);
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {page} (items {page * perPage - perPage + 1} -{' '}
+        {page * perPage < 42 ? page * perPage : 42} of 42)
```

Comment:
Create variable for this calculations(`page * perPage - perPage + 1`) and ternary operators(`page * perPage < 42 ? page * perPage : 42`). It will be easier for understanding